framework/app.mk

```diff
@@ -375,25 +385,25 @@ endif
 # for more explanations/details.
 uniq = $(if $1,$(firstword $1) $(call uniq,$(filter-out $(firstword $1),$1)))
 compilertype := unknown
-applibs :=  $(app_test_LIB) $(app_LIBS) $(depend_test_libs)
-applibs := $(call uniq,$(applibs))
-ifeq ($(libmesh_static),yes)
-  ifneq (,$(findstring clang,$(CXX)))
+ifneq (,$(findstring clang,$(CXX)))
```

Comment:
I'm really confused with what's going on in this section. First, do we have any raw compilers still compiling MOOSE? That seems... bad. Did something prompt all these changes?

Reply:
Is it not better to just fully dump something like
```
$CXX --version
```

Reply:
👍